app/src/main/java/gifapp/jsm/myapp/app/interview/jsmtech/gifapp/WebViewClass.java

```diff
@@ -0,0 +1,51 @@
+package gifapp.jsm.myapp.app.interview.jsmtech.gifapp;
+
+import android.os.Bundle;
+import android.support.v4.view.GestureDetectorCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.view.MotionEvent;
+import android.webkit.WebView;
+
+import helper.Gestures;
+
+/**
+ * Created by Sukriti on 7/26/16.
+ */
+public class WebViewClass extends AppCompatActivity {
+
+    String gifUrl = "";
+    private WebView mWebview;
+
+    private GestureDetectorCompat gestureDetectorCompat;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.web_view);
+
+        mWebview = (WebView) findViewById(R.id.web_view);
+        gestureDetectorCompat = new GestureDetectorCompat(this, new Gestures(WebViewClass.this));
+        // fetching url from the previous activity of the gif clicked
+        Bundle data = getIntent().getExtras();
+        if (data != null) {
+            gifUrl = data.getString("url");
+            mWebview.loadUrl(gifUrl);
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
```

Comment:
This logic is similar to the one in Gestures. As mentioned in another comment, we should have this class implement a IGestureListener interface and put that logic into IGestureListener.handleFling( )